.github/workflows/generation.yml

```diff
@@ -0,0 +1,63 @@
+name: Type Generation
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```

Comment:
```suggestion
```

Since `main` is protected I don't think this will succeed in committing changes there.

Which also means you should merge `main` into this branch, which is missing the TS version of TextInput